clients/src/main/java/org/apache/kafka/common/utils/Bytes.java

```diff
@@ -0,0 +1,415 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.utils;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Comparator;
+
+/**
+ * Utility class that handles byte arrays.
+ *
+ * Its implementation is inspired by Apache Hive, org.apache.hadoop.hbase.util.Bytes
+ */
+public class Bytes {
```

Comment:
As @ijuma said, there are quite a lot of methods in this class for which (afaiu) we don't have any immediate use.  We also don't have any unit tests for this class, so keep in mind that we will need to add such unit tests, and then we'll also need to add unit tests for methods that we aren't really using yet.